src/Features/CSharp/Portable/Snippets/ConsoleSnippetProvider.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.CompletionProviders.Snippets
+{
+    [Export(typeof(ISnippetProvider)), Shared]
+    internal class ConsoleSnippetProvider : AbstractSnippetProvider
```

Comment:
should likely have C# in the name.  Do we intend to write the snippets twice per langauge?

Reply:
I would think yes. Consider how so many of our codefixes support vb and c# and can work in both thanks to abstractions like SyntaxGenerator

Reply:
Yeah that makes sense, I can abstract it out